dlt/destinations/impl/sink/sink.py

```diff
@@ -0,0 +1,220 @@
+from abc import ABC, abstractmethod
+from types import TracebackType
+from typing import ClassVar, Dict, Optional, Sequence, Type, Iterable, Iterable
+
+from dlt.destinations.job_impl import EmptyLoadJob
+from dlt.common.typing import TDataItems
+from dlt.common import json
+
+from dlt.common.storages.load_storage import ParsedLoadJobFileName
+from dlt.common.schema import Schema, TTableSchema, TSchemaTables
+from dlt.common.schema.typing import TTableSchema
+from dlt.common.storages import FileStorage
+from dlt.common.destination import DestinationCapabilitiesContext
+from dlt.common.destination.reference import (
+    TLoadJobState,
+    LoadJob,
+    JobClientBase,
+)
+
+from dlt.destinations.impl.sink import capabilities
+from dlt.destinations.impl.sink.configuration import SinkClientConfiguration, TSinkCallable
+
+
+# TODO: implement proper state storage somewhere, can this somehow go into the loadpackage?
+job_execution_storage: Dict[str, int] = {}
+
+
+class SinkLoadJob(LoadJob, ABC):
+    def __init__(
+        self,
+        table: TTableSchema,
+        file_path: str,
+        config: SinkClientConfiguration,
+        schema: Schema,
+        job_execution_storage: Dict[str, int],
+    ) -> None:
+        super().__init__(FileStorage.get_file_name_from_file_path(file_path))
+        self._file_path = file_path
+        self._config = config
+        self._table = table
+        self._schema = schema
+        self._job_execution_storage = job_execution_storage
+
+        # TODO: is this the correct way to tell dlt to retry this job in the next attempt?
```

Comment:
read the docstrings for LoadJob. particular instance will not be retried. "retry" is a terminal state for the `LoadJob` class. loader.py will create a new instance of SinkLoadJob will be created